MyOwnORM/MyOwnORM/CustomDbSetReflection.cs

```diff
@@ -0,0 +1,693 @@
+            foreach (var str in strs)
+                res.Add($"INSERT INTO {tableName} VALUES ({str})");
+
+            return res.ToArray();
+        }
+
+        private protected static string MapEntityPropertyValuesInUpdateString(T obj, string idProperty)
+        {
+            Type type = obj.GetType();
+
+            PropertyInfo[] properties = type.GetProperties();
+            StringBuilder strBuilder = new StringBuilder();
+
+            for (int i = 0; i < properties.Length; i++)
+            {
+                PropertyInfo prop = properties[i];
+                string columnName = prop.Name;
+                object value = prop.GetValue(obj);
+
+                if (columnName == idProperty) continue;
+
+                if (prop.PropertyType.IsGenericType || (prop.PropertyType.IsClass && prop.PropertyType != typeof(string))) continue;
+
+                if (prop.PropertyType == typeof(string))
+                {
+                    strBuilder.Append($"{columnName}='{value}'");
+                }
+                else
+                {
+                    strBuilder.Append($"{columnName}={value}");
+                }
+
+                if (i < properties.Length - 1)
+                {
+                    strBuilder.Append(", ");
+                }
+            }
+
+            if (strBuilder.ToString().EndsWith(' '))
+            {
+                strBuilder.Length = strBuilder.Length - 2;
+            }
+
+            return strBuilder.ToString();
+        }
+
+        private protected static string MapEntityPropertyValuesInUpdateString(dynamic obj, string tableName, string idProperty, string idPropertyValue)
+        {
+            Type type = obj.GetType();
+
+            PropertyInfo[] properties = type.GetProperties();
+            StringBuilder strBuilder = new StringBuilder();
+
+            for (int i = 0; i < properties.Length; i++)
+            {
+                PropertyInfo prop = properties[i];
+                string columnName = prop.Name;
+
+                object value = prop.GetValue(obj);
+
+                if (columnName == idProperty) continue;
+
+                if (prop.PropertyType == typeof(string))
+                {
+                    strBuilder.Append($"{columnName}='{value}'");
+                }
+                else
+                {
+                    strBuilder.Append($"{columnName}={value}");
+                }
+
+                if (i < properties.Length - 1)
+                {
+                    strBuilder.Append(", ");
+                }
+            }
+
+            if (strBuilder.ToString().EndsWith(' '))
+            {
+                strBuilder.Length = strBuilder.Length - 2;
+            }
+
+            return $"UPDATE {tableName} SET {strBuilder} WHERE {idProperty}={idPropertyValue}";
+        }
+
+        private protected static string[] MapListEntitiesPropertyValuesInUpdateString(dynamic obj, string tableName, string idProperty, string idPropertyValue)
```

Comment:
don't use dynamic ever, this type breaks static typing and makes code less efficient. It is used rarely in cases when you need integration with dynamic-typing languages(like python). So instead of this use object(it is not good too), but the best solving of this problem are generics